pyteal/ast/ecdsa.py

```diff
@@ -111,33 +114,39 @@ def EcdsaVerify(
     return EcdsaVerifyExpr(curve, data, sigA, sigB, pubkey[0], pubkey[1])
 
 
-def EcdsaDecompress(curve: EcdsaCurve, compressed_pk: Expr) -> MultiValue:
+class EcdsaDecompress(MultiValue):
     """Decompress an ECDSA public key.
 
     Args:
         curve: Enum representing the ECDSA curve used for the public key
         compressed_pk: The compressed public key. Must be 33 bytes long and big endian encoded.
 
     Returns:
-        A MultiValue expression representing the two components of the public key, big endian
+        A MultiValue instance representing the two components of the public key, big endian
         encoded.
     """
 
-    if not isinstance(curve, EcdsaCurve):
-        raise TealTypeError(curve, EcdsaCurve)
+    def __init__(self, curve: EcdsaCurve, compressed_pk: Expr):
+        if not isinstance(curve, EcdsaCurve):
+            raise TealTypeError(curve, EcdsaCurve)
 
-    require_type(compressed_pk, TealType.bytes)
-    return MultiValue(
-        Op.ecdsa_pk_decompress,
-        EcdsaPubkey,
-        immediate_args=[curve.arg_name],
-        args=[compressed_pk],
-    )
+        self.curve = curve
 
+        require_type(compressed_pk, TealType.bytes)
+        super().__init__(
+            Op.ecdsa_pk_decompress,
+            EcdsaPubkey,
+            immediate_args=[curve.arg_name],
+            args=[compressed_pk],
+        )
 
-def EcdsaRecover(
-    curve: EcdsaCurve, data: Expr, recovery_id: Expr, sigA: Expr, sigB: Expr
-) -> MultiValue:
+    def __teal__(self, options: "CompileOptions"):
+        verifyFieldVersion(self.curve.arg_name, self.curve.min_version, options.version)
+
+        return super().__teal__(options)
+
+
+class EcdsaRecover(MultiValue):
     """Reover an ECDSA public key from a signature.
```

Comment:
@jdtzmn Minor:  Unrelated to the changeset, but easy to address the typo.

```suggestion
    """Recover an ECDSA public key from a signature.
```

Reply:
@jdtzmn Thanks for fixing the typo - resolving.